src/clients/packages_proposal.rb

```diff
@@ -234,6 +237,68 @@ def call_packageselector
 
         ret
       end
+
+      # Summary with with the list of uninstalled 3rd party packages
+      # @return [String,nil] Rich text summary or `nil` if no 3rd party package
+      #   is uninstalled
+      def orphaned_packages_warning
+        orphaned_packages = find_orphaned_packages
+        return nil if orphaned_packages.empty?
+
+        list = HTML.List(orphaned_packages_summary(orphaned_packages))
+        # TRANSLATORS: warning displayed in the upgrade summary, this informs the
+        # user that some manually installed non-SUSE packages will uninstalled,
+        # user should check if that is OK, it is possible to manually change the
+        # package status and keep it in the system
+        _("<b>Warning: These packages will be removed:</b> %s") % list
+      end
+
+      # find the orphaned non-SUSE packages which will be uninstalled from the system
+      # @return [Y2Packager::Resolvable] list of orphaned packages
+      def find_orphaned_packages
+        # preload the "vendor" attribute so the vendor matching below is faster
+        orphaned = Y2Packager::Resolvable.find(
+          { kind: :package, status: :removed, orphaned: true },
+          [:vendor]
+        )
+        # ignore SUSE or openSUSE packages, but include packages from OBS projects
+        # where vendor is like "obs://build.opensuse.org/YaST",
+        # the SUSE Hub packages use the "openSUSE" vendor
+        orphaned.reject! { |o| o.vendor.start_with?("SUSE") || o.vendor.start_with?("openSUSE") }
+        # sort the packages alphabetically by name for easier reading (case insensitive)
+        orphaned.sort_by! { |o| o.name.downcase }
+        log.info "Found #{orphaned.size} non-SUSE orphaned packages: " +
+          orphaned.map { |p| package_label(p) }.to_s
+        orphaned
+      end
+
+      # limit the list of the displayed orphaned packages, the list might be potentially huge,
+      # let's display just first few packages in the summary, the full list can be
+      # displayed in the package manager
+      ORPHANED_MAX_SIZE = 10
```

Comment:
NP: I would freeze the constant and mark it as private. BTW, shouldn't these new method be declared as private as well?